snapcraft/internal/meta/_errors.py

```diff
@@ -60,3 +61,17 @@ class AdoptedPartNotParsingInfo(SnapMetaGenerationError):
 
     def __init__(self, part: str) -> None:
         super().__init__(part=part)
+
+
+class AmbiguousPassthroughKeyError(SnapMetaGenerationError):
+
+    fmt = (
+        "Failed to propagate keys to snap metadata: "
+        "The following keys are specified as both regular keys and in "
+        "passthrough:\n"
+        '{keys}\n\n'
+        "Remove duplicate keys."
```

Comment:
This formatting isn't quite right:

"Failed to propagate keys to snap metadata: The following keys are specified as both regular keys and in passthrough:
grade, confinement

Remove duplicate keys."

How about using `formatting_utils.humanize_list`? Something like

```
fmt = (
    "Failed to generate snap metadata: "
    "The following keys are specified in their regular location as well as in passthrough: {}. "
    "Remove duplicate keys."
)

def __init__(self, keys: Set[str]) -> None:
        super().__init__(keys=formatting_utils.humanize_list(keys, 'and')
```